src/torrent.cpp

```diff
@@ -2330,11 +2331,11 @@ namespace libtorrent
 
 		state_updated();
 
-		if (m_resume_data && m_resume_data->node.type() == bdecode_node::dict_t)
+		if (resume_data && resume_data->node.type() == bdecode_node::dict_t)
```

Comment:
I think a lot, if not all, of this is done earlier in master. this change seems likely to conflict with master (i.e. difficult to bring forward)